src/main/kotlin/no/nav/familie/ef/mottak/service/MappeService.kt

```diff
@@ -0,0 +1,111 @@
+package no.nav.familie.ef.mottak.service
+
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.nav.familie.ef.mottak.integration.IntegrasjonerClient
+import no.nav.familie.ef.mottak.repository.domain.Søknad
+import no.nav.familie.ef.mottak.service.MappeSøkestreng.OPPLÆRING
+import no.nav.familie.ef.mottak.service.MappeSøkestreng.SELVSTENDIG
+import no.nav.familie.ef.mottak.service.MappeSøkestreng.SÆRLIG_TILSYNSKREVENDE
+import no.nav.familie.ef.mottak.service.MappeSøkestreng.UPLASSERT
+import no.nav.familie.ef.mottak.util.dokumenttypeTilStønadType
+import no.nav.familie.kontrakter.ef.søknad.Aktivitet
+import no.nav.familie.kontrakter.ef.søknad.SøknadBarnetilsyn
+import no.nav.familie.kontrakter.ef.søknad.SøknadOvergangsstønad
+import no.nav.familie.kontrakter.ef.søknad.Søknadsfelt
+import no.nav.familie.kontrakter.felles.ef.StønadType
+import no.nav.familie.kontrakter.felles.objectMapper
+import no.nav.familie.kontrakter.felles.oppgave.FinnMappeRequest
+import no.nav.familie.kontrakter.felles.oppgave.FinnMappeResponseDto
+import no.nav.familie.kontrakter.felles.oppgave.MappeDto
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+
+@Service
+class MappeService(
+    private val integrasjonerClient: IntegrasjonerClient,
+    private val søknadService: SøknadService
+) {
+
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+
+    fun finnMappeIdForSøknadOgEnhet(søknadId: String, enhetsnummer: String?): Long? {
+        if (enhetsnummer != OppgaveService.ENHETSNUMMER_NAY) {
+            return null
+        }
+        val mapper = finnMapperForEnhet(enhetsnummer)
+        val søkestreng = finnSøkestrengForSøknad(søknadId)
+        return utledMappeForSøkestreng(mapper, søkestreng)?.id?.toLong()
+    }
+
+    private fun finnMapperForEnhet(enhetsnummer: String): FinnMappeResponseDto {
+        val finnMappeRequest = FinnMappeRequest(
+            tema = listOf(),
+            enhetsnr = enhetsnummer,
+            opprettetFom = null,
+            limit = 1000
+        )
+        val mapperResponse = integrasjonerClient.finnMappe(finnMappeRequest)
+        log.info("Mapper funnet: Antall: ${mapperResponse.antallTreffTotalt}, ${mapperResponse.mapper} ")
```

Comment:
Trenger vi å logge dette for hvert tilfelle vi henter mapper for enhet? 
https://logs.adeo.no/app/discover#/doc/96e648c0-980a-11e9-830a-e17bbd64b4db/logstash-apps-prod-005395?id=1cwcE4MBYoaSJGu9dM2x

Reply:
Fint å logge, men nå logges det kun når vi henter fra integrasjoner. Cachen logger ikke.